server/src/main/java/org/elasticsearch/index/IndexingPressure.java

```diff
@@ -203,6 +256,10 @@ public Releasable markPrimaryOperationStarted(int operations, long bytes, boolea
             this.currentCombinedCoordinatingAndPrimaryBytes.getAndAdd(-bytes);
             this.primaryRejections.getAndIncrement();
             this.primaryDocumentRejections.addAndGet(operations);
+            if (operationExpansionTracking) {
```

Comment:
Hmm. Does this make sense? We've already pass the "large document" check. And this just means we passed the primary limit when attempting to do the expansion.

Reply:
I kind of feel like if we get to this point and it is not a "large document" this should count as a primary rejection only as it might be just a normal sized document.

However, I'm open to changing this if there is a logical reason from an autoscaling perspective why we want to treat this.